qiskit/providers/ibmq/jupyter/dashboard/dashboard.py

```diff
@@ -0,0 +1,312 @@
+        """
+        for _wid in self.dashboard._device_list.children:
+            _wid.close()
+        self.dashboard._device_list.children = []
+        for back in self.backend_dict.values():
+            _thread = threading.Thread(target=_add_device_to_list,
+                                       args=(back, self.dashboard._device_list))
+            _thread.start()
+
+    def start_dashboard(self):
+        """Starts the job viewer
+        """
+        self.dashboard = build_dashboard_widget()
+        self.job_viewer = self.dashboard.children[0].children[1]
+        self._get_backends()
+        self.refresh_device_list()
+        self.dashboard._thread = threading.Thread(target=update_backend_info,
+                                                  args=(self.dashboard._device_list,))
+        self.dashboard._thread.do_run = True
+        self.dashboard._thread.start()
+        self.refresh_jobs_board()
+
+    def stop_dashboard(self):
+        """Stops the job viewer.
+        """
+        if self.dashboard:
+            self.dashboard._thread.do_run = False
+            self.dashboard._thread.join()
+            self.dashboard.close()
+        self.dashboard = None
+
+    def update_single_job(self, update_info: Tuple):
+        """Update a single job instance
+
+        Args:
+            update_info (tuple): Updated job info.
+        """
+        job_id = update_info[0]
+        found_job = False
+        ind = None
+        for idx, job in enumerate(self.jobs):
+            if job.job_id == job_id:
+                found_job = True
+                ind = idx
+                break
+        if found_job:
+            job_wid = self.jobs[ind]
+            # update status
+            if update_info[1] == 'DONE':
+                stat = "<font style='color:#34BC6E'>{}</font>".format(update_info[1])
+            elif update_info[1] == 'ERROR':
+                stat = "<font style='color:#DC267F'>{}</font>".format(update_info[1])
+            elif update_info[1] == 'CANCELLED':
+                stat = "<font style='color:#FFB000'>{}</font>".format(update_info[1])
+            else:
+                stat = update_info[1]
+            job_wid.children[3].value = stat
+            # update queue
+            if update_info[2] == 0:
+                queue = '-'
+            else:
+                queue = str(update_info[2])
+            job_wid.children[4].value = queue
+
+    def cancel_job(self, job_id: str):
+        """Cancels a job in the watcher
+
+        Args:
+            job_id: Job id to remove.
+
+        Raises:
+            Exception: Job id not found.
+        """
+        do_pop = False
+        ind = None
+        for idx, job in enumerate(self.jobs):
+            if job.job_id == job_id:
+                do_pop = True
+                ind = idx
+                break
+        if not do_pop:
+            raise Exception('job_id not found')
+        if 'CANCELLED' not in self.jobs[ind].children[3].value:
```

Comment:
Shouldn't this check for DONE and ERROR in addition to CANCELLED? Not that it matters much since the `cancel()` will just fail.

Reply:
This is just trying to cancel a job after you clicked the cancel button.

Reply:
But presumably you're doing `if 'CANCELLED` because canceling is moot if the job has already been cancelled? That applies to `DONE` and `ERROR` as well.

Reply:
Ok yes, I see what your saying now.